### ISSUE TITLE
a11y: real time test progress is not announced to screen readers

### ISSUE THREAD

**Issue by M-V-RAGHUPATHI-SAI** [opentechinstitute/mlab-speedtest#205]
## Description
The dynamic speed updates currently lack aria live region, preventing them from being announced to assistive technologies as required by WCAG.

## Solution
I can fix this by adding `aria-live="polite"` to the progress-label class.

**Comment by robertodauria** [comment]
Thanks for reporting this. I agree, just noting that this change will require thorough testing with actual screen readers. In particular I'm wondering if `polite` is still going to cause announcements. Is 10s enough time for multiple announcements? Could a simple "Test running, please wait"/"Download test completed, now running upload test" work better in this case?

**Comment by M-V-RAGHUPATHI-SAI** [comment]
Thanks for the feedback, that's actually a great point, `polite` will likely queue up all the speed updates and may not finish it in 10s and users may feel like overload. i agree with your approach is a better UX, announcing only the changes gives them context about what's going on.

We can use aria-live only for phase changes like you mentioned, I plan to match our existing caching pattern, i am thinking to add a visually hidden aria-live region in the HTML, and update its content using i18n whenever the test phase changes.

We can keep it simple and at the start "test started" and at the end we can say "test completed", we can go complex and mention "download started" or "upload started", because we have to cover this in 10s, i am thinking to go simpler what do you think? 

I will test the final flow locally with NVDA to make sure it sounds natural and report back once we finalized the flow.

**Comment by M-V-RAGHUPATHI-SAI** [comment]
"Test started", "Download test completed, now running upload test", and finally "test completed" also sounds fine to me. what do you think??

Also i have read that `display: none` completely hides the element from screen readers. so, in accordance with WCAG guidelines, i found this CSS for visually hiding content while keeping it accessible to screen readers, which makes the text to 1px box:

```
.visually-hidden {
    position: absolute;
    width: 1px;
    height: 1px;
    margin: -1px;
    padding: 0;
    overflow: hidden;
    clip: rect(0, 0, 0, 0);
    clip-path: inset(50%);
    white-space: nowrap;
    border: 0;
}
```

>"Test started", "Download test completed, now running upload test", and finally "test completed" also sounds fine to me. what do you think??

Once we are confirmed with this, i will start working on the PR.

**Comment by M-V-RAGHUPATHI-SAI** [comment]
First of all, thanks for the guidance till now @robertodauria. 

Though M-Lab has dropped from GSoC 2026, i would still love to work with M-lab, as they have mentioned in the mail to contribute after gsoc deadline, i will be coming back to work on this issue after the deadline.
